ambry-quota/src/main/java/com/github/ambry/quota/MaxThrottlePolicy.java

```diff
@@ -24,18 +25,52 @@
  * groups the quota usage for all the quotas.
  */
 public class MaxThrottlePolicy implements ThrottlePolicy {
+  static final long DEFAULT_RETRY_AFTER_MS = ThrottlingRecommendation.NO_RETRY_VALUE_FOR_RETRY_AFTER_MS;
+  static final int DEFAULT_RECOMMENDED_HTTP_STATUS = HttpResponseStatus.OK.code();
+  static final QuotaWarningLevel DEFAULT_QUOTA_WARNING_LEVEL = QuotaWarningLevel.HEALTHY;
+  static final boolean DEFAULT_SHOULD_THROTTLE = false;
+
+  // Percentage usage at or below this limit is healthy.
+  static final int HEALTHY_USAGE_LEVEL_LIMIT = 80;
+  // Percentage usage level at or below this limit and above healthy limit will generate warning.
+  static final int WARNING_USAGE_LEVEL_LIMIT = 95;
+  // Percentage usage level at or below this limit and above warning limit will generate critical warning. Anything above will be fatal.
+  static final int CRITICAL_USAGE_LEVEL_LIMIT = 100;
+
   @Override
   public ThrottlingRecommendation recommend(List<QuotaRecommendation> quotaRecommendations) {
     boolean shouldThrottle = false;
     Map<QuotaName, Float> quotaUsagePercentage = new HashMap<>();
-    int recommendedHttpStatus = 200;
-    long retryAfterMs = -1;
+    int recommendedHttpStatus = DEFAULT_RECOMMENDED_HTTP_STATUS;
+    long retryAfterMs = DEFAULT_RETRY_AFTER_MS;
+    QuotaWarningLevel quotaWarningLevel = DEFAULT_QUOTA_WARNING_LEVEL;
     for (QuotaRecommendation recommendation : quotaRecommendations) {
       shouldThrottle = shouldThrottle | recommendation.shouldThrottle();
       quotaUsagePercentage.put(recommendation.getQuotaName(), recommendation.getQuotaUsagePercentage());
       recommendedHttpStatus = Math.max(recommendation.getRecommendedHttpStatus(), recommendedHttpStatus);
       retryAfterMs = Math.max(recommendation.getRetryAfterMs(), retryAfterMs);
+      quotaWarningLevel = QuotaWarningLevel.fromInt(Math.max(quotaWarningLevel.ordinal(),
+          computeWarningLevel(recommendation.getQuotaUsagePercentage()).ordinal()));
+    }
```

Comment:
Java enum has already built-in `compareTo()` method, which uses the enum position (ordinal) to compare one object to other.
So I guess this can be simplified to:
```
quotaWarningLevel = Math.max(quotaWarningLevel, computeWarningLevel(recommendation.getQuotaUsagePercentage()));
```